### ISSUE TITLE
Add `maybe_clamp` to `maybe_math`

### ISSUE THREAD

**Issue by Weibye** [DioxusLabs/taffy#247]
Several places in our code we are doing the following pattern:
```rs
.maybe_max(node_min_size.width)
.maybe_min(node_max_size.width),
```
This can instead be replaced with a `maybe_clamp` to better capture what is happening and why. 

> Maybe we should introduce a `maybe_clamp`, this pattern seems to be common.
It could probably be a default implementation for the trait.

_Originally posted by @TimJentzsch in https://github.com/DioxusLabs/taffy/pull/246#discussion_r1027102142_

**Comment by nicoburns** [comment]
I've implemented this on the branch for #246. Normally, I'd do a separate PR, but #246 implements `MaybeMath` for `AvailableSpace`, and I don't want to lose those impls.

**Comment by alice-i-cecile** [comment]
Yep I'm on board with that strategy.
